x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/utils.ts

```diff
@@ -193,6 +195,134 @@ export const getTupleDuplicateErrorsAndUniqueRules = (
   return [Array.from(errors.values()), Array.from(rulesAcc.values())];
 };
 
+/**
+ * Query for a saved object with a given origin id and replace the
+ * id in the provided action with the _id from the query result
+ * @param action
+ * @param esClient
+ * @returns
+ */
+export const swapActionIds = async (
+  action: Action,
+  esClient: ElasticsearchClient
+): Promise<Action | Error> => {
+  try {
+    // actions are hidden SO types so we need to query
+    // with the elasticsearch client.. unless there
+    // is a better way to do this..
```

Comment:
You can create a regular SOClient and specify `includedHiddenTypes: ['action']`:

https://github.com/elastic/kibana/blob/f7163878c0f291ab11d2b3daa7ec9df19786198d/src/core/server/saved_objects/saved_objects_service.ts#L173-L185

https://github.com/elastic/kibana/blob/4584a8b570402aa07832cf3e5b520e5d2cfa7166/src/core/server/saved_objects/service/lib/scoped_client_provider.ts#L57-L60

You should really be using the SOClient because it is space-aware :+1:

Reply:
Ah!!! Ok perfect. Thank you so much! I definitely did not want to use the ES client. Missed this setting.